### PR TITLE
Move Learn Pulumi to Pulumi Tutorials

### PR DESCRIPTION
**Pull request by cnunciato**
This change creates a new section called Pulumi Tutorials (🎉) and bootstraps it with content from Learn Pulumi. As part of this work, I also reviewed (and fixed!) all existing content and redirected all content from `/learn` to its new home at `/tutorials`. 

A few things I've not yet done:

* Documented how to create a new tutorial, how tutorials are sorted, grouped, sorted within a group, etc.
* ~~Added a `make` helper/Hugo archetype to simplify creating a new tutorial. We have these for Learn today and they're nice, but they don't map directly to the new stuff, so they'll need some adjusting.~~
* Documented the concept of "collections" as it applies to tutorials. Everything here is a collection, and collections are typed -- so there are "provider" collections (e.g., `aws`), "proficiency" (i.e., skill-level) collections (`pulumi-101`), there'll surely be "product" and "feature" collections soon (ESC, Deployments, Components), etc. Collections are the only grouping mechanism we have, but the `type` field should make this pretty flexible (certainly flexible enough for now), and we can always extend/refactor the content model when/if we outgrow this. Tutorials can also belong to multiple collections, of course, or none, in which case they'll spill into the All Tutorials section on the Tutorials home page.

    Since there was so little content to go live with here, I created a single collection to start, and called it "Pulumi 101" (feel free to suggest an alternative -- I'm mainly looking for something succinct that conveys that these are meant for new users). It contains the first three Learn modules --- Fundamentals, Building, and Encapsulation, since those are all about core concepts (whereas Automation API and Importing are more advanced topics).
* Written up authoring guidelines for tutorials -- what makes a good one, when to use multi-page vs. single-page, a general a structure to follow for both, etc.
* Consolidated/figured out what to do with the "[cloud engineering glossary](https://www.pulumi.com/learn/glossary/)". It does feel like there's a place for a glossary of sorts in this new section, so I'll probably find a way to integrate it somehow, but for now, it's got the same treatment it does today (and is redirected to /tutorials/glossary like everything else).  
* ~~Made all the pre-reqs and intros consistent. I'll be doing that while y'all review what's here.~~

It may also feel like the navigation is a bit "sparse" right now, and that's mainly because we have so little content. As we add more tutorials, collections, and collection types, this will naturally grow and begin got look more like what we've been aiming for.

Fixes https://github.com/pulumi/devrel-team/issues/641
Fixes https://github.com/pulumi/docs/issues/12017
Fixes https://github.com/pulumi/docs/issues/10226
Fixes https://github.com/pulumi/docs/issues/10225
Fixes https://github.com/pulumi/docs/issues/12418
Fixes https://github.com/pulumi/docs/issues/10619
Fixes https://github.com/pulumi/docs/issues/10202
Fixes https://github.com/pulumi/docs/issues/10173
Fixes https://github.com/pulumi/docs/issues/10163
Fixes https://github.com/pulumi/docs/issues/10155
Fixes https://github.com/pulumi/docs/issues/11688
Fixes https://github.com/pulumi/docs/issues/12423
Fixes https://github.com/pulumi/docs/issues/12426

### CODE REVIEW

**Comment by sean1588** [comment]
Looks good overall. Couple things I noticed here with regards to how some things are rendering:

**It looks like the left nav is cutting off some of the elements within it.**
Screen Shot 2024-08-06 at 3 58 24 PM
Screen Shot 2024-08-06 at 3 57 45 PM

**I think we could do with a decrease of padding between the boundaries of the left nav and the menu itself. It seems to be less on other pages**
Screen Shot 2024-08-06 at 3 57 54 PM

**Review by interurban — APPROVED** [review]
Content looks good, I'll let Sean give the the code/approval.

File: Makefile [hunk 1]
```diff
@@ -116,13 +108,17 @@ test-programs:
 upgrade-programs:
 	./scripts/programs/upgrade.sh
 
-.PHONY: new-learn-module
-new-learn-module:
-	./scripts/content/new-learn-module.sh
+.PHONY: new-tutorial-module
```
Comment (sean1588) [inline Makefile:111]: good idea!!

**Review by sean1588 — APPROVED** [review]
I'm going to go ahead and leave you the approval here, assuming we get the nav sorted. There are a few more things I noticed when checking out the mobile view though. Not sure if all of these are blocking though, maybe we can fast follow and address these things after.

looks like the bullet points are not wrapping the text
Screen Shot 2024-08-07 at 3 13 52 PM

these look like they need some spacing and alignment adjustment
Screen Shot 2024-08-07 at 3 14 19 PM

the menu seems to be aligned all the way to the right, not sure if that was intentional though.
Screen Shot 2024-08-07 at 3 13 09 PM

**Comment by sean1588** [comment]
one more thing, I noticed the search bar is not aligned to the right anymore
Screen Shot 2024-08-07 at 3 24 39 PM

**Comment by sean1588** [comment]
🚀

**Pull request merged**